src/main/scala/is/hail/expr/HailRep.scala

```diff
@@ -2,8 +2,9 @@ package is.hail.expr
 
 import is.hail.utils.Interval
 import is.hail.variant.{AltAllele, Call, Genotype, Locus, Variant}
+import scala.collection.mutable
 
-trait HailRep[T] { self =>
+trait HailRep[-T] { self =>
```

Comment:
I don't think this is right either.  HailReps are used on both inputs and outputs for functions registered in the function registry, so wether they should be treated co- or contravariantly depends on the context.  Or am I confused?